src/sql_mock/table_mocks.py

```diff
@@ -164,13 +193,18 @@ def as_sql_input(self):
             str: A SQL query that combines data from all rows.
         """
         # Convert the instance into a SQL snippet for CTE input
-        if len(self._data) == 0:
+        if len(self._sql_mock_data.data) == 0:
             # Populate default values row with a WHERE FALSE statement to simulate no rows for the model
             snippet = self._to_sql_row({})
             snippet += " WHERE FALSE"
         else:
-            snippet = "\nUNION ALL\nSELECT ".join([self._to_sql_row(row_data) for row_data in self._data])
-        return f"{self._table_ref} AS (\n" f"SELECT {snippet}\n" ")"
+            snippet = "\nUNION ALL\nSELECT ".join(
+                [self._to_sql_row(row_data) for row_data in self._sql_mock_data.data]
+            )
+
+        # Indent whole CTE content for better query readability
```

Comment:
Wonderful 😄 a query with good readability built from various blocks of code is high-quality stuff!